Core/src/com/infiniteautomation/mango/spring/service/ConfigurationTemplateService.java

```diff
@@ -0,0 +1,167 @@
+/*
+ *  Copyright (C) 2023 RadixIot LLC. All rights reserved.
+ */
+
+package com.infiniteautomation.mango.spring.service;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Service;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import com.serotonin.m2m2.Common;
+import com.serotonin.m2m2.vo.permission.PermissionException;
+
+import au.com.bytecode.opencsv.CSVReader;
+
+@Service
+public class ConfigurationTemplateService {
```

Comment:
Service should have an interface to allow for easy mocking during testing

Reply:
Interface implemented.